keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -84,10 +85,11 @@ public static Nd4jDoubleTensor arange(double start, double end, double stepSize)
     private INDArray tensor;
 
     public Nd4jDoubleTensor(double[] data, int[] shape) {
-        this.tensor = TypedINDArrayFactory.create(data, shape, BUFFER_TYPE);
+        this(TypedINDArrayFactory.create(data, shape, BUFFER_TYPE));
```

Comment:
Does ND4J throw an exception if the shape is zero length? If it doesn't throw an exception and returns a 1x1 then we might want to do the check before the ND4J INDArray creation.

Reply:
ND4J doesn't throw, it returns an INDArray whose shape is an empty `int[]`.